src/engine/thread.js

```diff
@@ -186,7 +183,25 @@ class Thread {
          */
         this.warpTimer = null;
 
+        /**
+         * The value just reported by a promise.
+         * @type {*}
+         */
         this.justReported = null;
+
+        /**
+         * The id of the block that we will report the promise resolved value
+         * for.
+         * @type {string}
+         */
+        this.reporting = null;
```

Comment:
Nit: Can we rename this to `this.reportingBlock` for clarity? As is, the name could be mistaken for a boolean.

Reply:
I changed it to `reportingBlockId`.